features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/timeline/components/TimelineItemEventRow.kt

```diff
@@ -590,3 +643,83 @@ internal fun TimelineItemEventRowPreview() = ElementPreview {
         }
     }
 }
+
+@Preview(
+    name = "Encryption Shields"
+)
+@Preview(
+    name = "Encryption Shields - Night",
+    uiMode = Configuration.UI_MODE_NIGHT_YES
+)
+@Composable
+internal fun TimelineItemEventRowShieldsPreview() = ElementPreview {
+    Column {
+        ATimelineItemEventRow(
+            event = aTimelineItemEvent(
+                senderDisplayName = "Sender with a super long name that should ellipsize",
+                isMine = true,
+                content = aTimelineItemTextContent(
+                    body = "Message sent from unsigned device"
+                ),
+                groupPosition = TimelineItemGroupPosition.First,
+                messageShield = aRedShield()
+            ),
+        )
+        ATimelineItemEventRow(
+            event = aTimelineItemEvent(
+                senderDisplayName = "Sender with a super long name that should ellipsize",
+                content = aTimelineItemTextContent(
+                    body = "Short Message with authenticity warning"
+                ),
+                groupPosition = TimelineItemGroupPosition.Middle,
+                messageShield = aGreyShield()
+            ),
+        )
+        ATimelineItemEventRow(
+            event = aTimelineItemEvent(
+                isMine = true,
+                content = aTimelineItemImageContent().copy(
+                    aspectRatio = 2.5f
+                ),
+                groupPosition = TimelineItemGroupPosition.Last,
+                messageShield = aRedShield()
+            ),
+        )
+        ATimelineItemEventRow(
+            event = aTimelineItemEvent(
+                content = aTimelineItemImageContent().copy(
+                    aspectRatio = 2.5f
+                ),
+                groupPosition = TimelineItemGroupPosition.Last,
+                messageShield = aGreyShield()
+            ),
+        )
+    }
+}
+
+@Composable
+@ReadOnlyComposable
+private fun TimelineItem.Event.shieldPosition(): MessageShieldPosition {
+    val shield = this.messageShield ?: return MessageShieldPosition.None
+
+    // sdk returns raw human readable strings, add i18n support
```

Comment:
@Jorge Could the SDK return an enum instead?

Reply:
I think you should discuss this with @pixlwave (and not poor `@jorge` 😅 ).

Reply:
Ah yeah I fixed this… You can now switch on the `ShieldState::code` which is an enum 😁

On EXI we map `ShieldState` to this [`EncryptionAuthenticity`](https://github.com/element-hq/element-x-ios/blob/develop/ElementX/Sources/Services/Timeline/TimelineItemContent/EncryptionAuthenticity.swift) representation for our sanity.

Reply:
@bmarty For completeness here is the view we use for the timestamp with send/encryption warnings: [`TimelineItemSendInfoLabel`](https://github.com/element-hq/element-x-ios/blob/develop/ElementX/Sources/Screens/Timeline/View/Style/TimelineItemSendInfoLabel.swift)

And the PRs https://github.com/element-hq/element-x-ios/pull/3051, https://github.com/element-hq/element-x-ios/pull/3100 and https://github.com/element-hq/element-x-ios/pull/3116

Reply:
> Same for the color, it should be a enum (if the first enum is not enough) with a notion of criticity. "color" is a concept for the application (UI) level.

FWIW, both myself and @stefanceriu raised this [here](https://github.com/element-hq/element-x-ios/pull/3051#discussion_r1680882169) and [here](https://github.com/element-hq/element-x-ios/pull/3116#discussion_r1705128756) but the crypto team don't want to change it 🤷‍♂️